### PR TITLE
Update collab frontend to send req to history service

### PR DESCRIPTION
**Pull request by qinxutan**
This PR does the following as requested:
1) upon routing to collab page and on “sessionData” event, send a fetch request to history service. data to be sent includes userUid, questionUid, dateAttempted. Use moment().tz("Asia/Singapore").format(); to get the dateAttempted in frontend, and store it in the frontend.
 
2} upon running code, send a fetch request to history_service alongside user executed code whenever user executes code. The data to be sent over are userUid, questionUid, writtenCode, dateAttempted store in (1).

shown here on console it is sent to history svc correctly and also reflected on history svc page correctly:
 
AF8F9628-F6E0-4F3E-99BE-544B95A0C954

0A2E20BB-AAA9-444B-9D2A-69DF44DB5534

### CODE REVIEW

File: frontend/src/views/CollabPageView.tsx [hunk 1]
```diff
@@ -117,11 +119,53 @@ const CollabPageView: React.FC = () => {
...
+            console.log(result.message);
+          } else {
+              const error = await response.json();
+              console.error('Error creating question:', error.message);
+          }
+        } catch (error) {
+          console.error('Network error:', error);
+        }
+
       });
```
Comment (Tan-Jia-Rong) [inline frontend/src/views/CollabPageView.tsx]: Consider using `callFunction` for this

File: frontend/src/views/CollabPageView.tsx [hunk 2]
```diff
@@ -267,6 +311,50 @@ const CollabPageView: React.FC = () => {
...
+      console.log("Request URL for storing executed code:", storeUserExecutedCode);
+
+      const result = await fetch(storeUserExecutedCode, {
+        method: "POST",
+        headers: {
+            'Authorization': `Bearer ${token}`, // If using JWT or similar
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(requestBody),
+      });
```
Comment (Tan-Jia-Rong) [inline frontend/src/views/CollabPageView.tsx]: This too. consider using call function

File: .env [hunk 3]
```diff
@@ -9,3 +9,5 @@ VITE_USER_SERVICE_BACKEND_URL=http://localhost:5001
 VITE_COLLAB_SERVICE_WS_BACKEND_URL=ws://localhost:5004
 VITE_COLLAB_SERVICE_BACKEND_URL=http://localhost:5004
 VITE_HISTORY_SERVICE_BACKEND_URL=http://localhost:5005
+VITE_HISTORY_SERVICE_CREATE_QUESTION_BACKEND_URL=http://localhost:5005/create-question-attempted
+VITE_HISTORY_SERVICE_STORE_USER_CODE_BACKEND_URL=http://localhost:5005/store-user-executed-code
```
Comment (Tan-Jia-Rong) [inline .env]: @ekweirui do we do it as such all do we pass it in the form of

`url, functionName,other params i dont rmb`

**Review by Tan-Jia-Rong — COMMENTED** [review]
minor comments. But generally lgtm :)

File: frontend/src/views/CollabPageView.tsx (hunk 1 above)
Comment (qinxutan) [reply frontend/src/views/CollabPageView.tsx]: oh tq have edited to use callfunction :-)

File: .env (hunk 3 above)
Comment (ekweirui) [reply .env]: I think we can delete this 2 env variables since we are not using them anywhere!

Comment (qinxutan) [reply .env]: yaa ok! have done so tq :)

File: frontend/src/views/CollabPageView.tsx [hunk 4]
```diff
@@ -117,11 +121,40 @@ const CollabPageView: React.FC = () => {
...
+        setDateAttempted(attemptDate);
+        console.log("Attempt date:", attemptDate);
+        
+        const requestBody = {
+          userUid: uid, 
+          questionUid: questionData.id, 
+          dateAttempted: attemptDate,
+        };
+
+        console.log("Request Body:", requestBody);
```
Comment (ekweirui) [inline frontend/src/views/CollabPageView.tsx:140]: Minor issue but would be good if `requestBody` is not logged so we don't flood our client's browser console.

File: frontend/src/views/CollabPageView.tsx [hunk 5]
```diff
@@ -267,6 +300,29 @@ const CollabPageView: React.FC = () => {
...
         : setIsError(false);
+
+      const requestBody = {
+        userUid: userId, 
+        questionUid: questionData.id, 
+        dateAttempted: dateAttempted,
+        codeWritten: sourceCode,
+      };
+
+      console.log("Request Body for storing executed code:", requestBody);
```
Comment (ekweirui) [inline frontend/src/views/CollabPageView.tsx:311]: Similar point to above

**Review by ekweirui — APPROVED** [review]
LGTM 👍 Thanks for implementing the collab frontend to history service functionality so smoothly! 💯

**Pull request merged**